### ISSUE TITLE
Problem compiling ZERO function on Linux. _fds_bits should set 32 bits.

### ISSUE THREAD

**Issue by jmarcelino** [IBM-Swift/BlueSocket#10]
Hi,

First let me say thank you. This is a really great project and love that it's already been updated to the latest Swift snapshot.

I'm trying to compile the latest release 0.7.1 on Debian Linux Jessie on armv7l using the latest Swift snapshot but compile fails with:

Socket-0.7.1/Sources/SocketUtils.swift:121:20: error: cannot assign value of type '(Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int)' to type '(__fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask, __fd_mask)' (aka '(Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int, Int)')
                set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)

Since _fd_mask is defined in sys/select.h as a long it should be 32 bits and sure enough replacing 
`set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)`
with 
`set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)`
solved the problem and everything compiled.

 I'm not sure why it's currently set to only 16 bits on Linux inside the #if os(Linux) block?

SET, CLR and ISSET need to be updated too and be very similar to the macOS version, minus the fd_set and __fd_set difference (perhaps there's a way to map that? my Swift/C integration-fu is not quite there yet)

**Comment by billabt** [comment]
Thanks for the kind comments.  But right now we're only supporting Linux on **ubuntu**. where although the __fd_mask is defined as a long fd_set.__fds_bits is only 16 not 32 bits.

Using the Swift REPL, you can see this very clearly...
```
Welcome to Swift version 3.0-dev (LLVM 440a472499, Clang e10506ae1c, Swift 395e967875). Type :help for assistance.
  1> import Glibc
  2>  
  3> let set = fd_set()
set: fd_set = {
  __fds_bits = {
    0 = 0
    1 = 0
    2 = 0
    3 = 0
    4 = 0
    5 = 0
    6 = 0
    7 = 0
    8 = 0
    9 = 0
    10 = 0
    11 = 0
    12 = 0
    13 = 0
    14 = 0
    15 = 0
  }
}
  4>  
```

**Comment by jmarcelino** [comment]
Fair enough, but to be more specific this means the code only supports Ubuntu on x86. 
Ubuntu on ARM  (for example on the Raspberry Pi 2) also expects 32 bits.

Personally I think this is an issue for users of Aphid, which - as you probably know - is the new MQTT Swift library also from IBM and which uses BlueSocket. Aphid/MQTT is ideal in low cost ARM targets to feed data for example to services like IBM Watson IoT... but that's just one application and I respect that you don't wish to support other targets at this time.

**Comment by rfdickerson** [comment]
@billabt we saw this, too, when @sandmman and I were compiling BlueSocket on the Pi (ARM 32-bits) for our Aphid project. Is there some way we can dynamically change fds_bits it so that it builds on the Pi?

**Comment by billabt** [comment]
I'm looking into it.  But at this point, I'm not sure we can accommodate it given the current toolset.  There's unfortunately no way to determine which Linux were running on.

**Comment by jmarcelino** [comment]
Perhaps testing for arch(arm)? 

Something like this works on both arm (Pi, Beagles, etc...) and x86 targets for me, but I'm not sure it holds true for everything. Still it could help.

#if arch(arm) && os(Linux) //Set 32 bits
set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
#elseif os(Linux) // Set 16 bits
set.__fds_bits = (0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0)
#endif

**Comment by billabt** [comment]
I'm looking at a potential solution right that should work on any Linux platform.  Will advise.

**Comment by rfdickerson** [comment]
Oh, I didn't realize that arch(arm) works. There's also `__WORDSIZE` that can get you back 32 and 64.

**Comment by billabt** [comment]
My plan didn't pan out...  I'm going to go with the arch(arm) solution.

**Comment by billabt** [comment]
This issue is fixed in version 0.7.2.  @jmarcelino your solution to use `arm` seems to be working.  Thanks for the suggestion.
